app/src/main/java/com/adammcneilly/toa/MainActivity.kt

```diff
@@ -100,6 +113,124 @@ class MainActivity : FragmentActivity() {
 
         val navController = navigationEngine.rememberNavController()
 
+        val windowSizeClass = calculateWindowSizeClass(activity = this)
+
+        val navigationType = getNavigationType(windowSizeClass.widthSizeClass)
+
+        val navigationTabs = listOf(
+            NavigationTab.Home,
+            NavigationTab.Settings,
+        )
+
+        when (navigationType) {
+            NavigationType.BOTTOM_NAVIGATION -> {
+                BottomNavigationContent(
+                    startRoute,
+                    navigationEngine,
+                    navController,
+                    windowSize,
+                    navigationTabs
+                )
+            }
+
+            NavigationType.NAVIGATION_RAIL -> {
+                NavigationRailContent(
+                    navController,
+                    navigationTabs,
+                    startRoute,
+                    navigationEngine,
+                    windowSize
+                )
+            }
+
+            NavigationType.PERMANENT_NAVIGATION_DRAWER -> {
+                PermanentNavigationDrawerContent(
+                    navController,
+                    navigationTabs,
+                    startRoute,
+                    navigationEngine,
+                    windowSize
+                )
+            }
+        }
+    }
+
+    @Composable
+    private fun PermanentNavigationDrawerContent(
+        navController: NavHostController,
+        navigationTabs: List<NavigationTab>,
+        startRoute: Route,
+        navigationEngine: NavHostEngine,
+        windowSize: WindowSize
+    ) {
+        PermanentNavigationDrawer(
+            drawerContent = {
+                TOANavigationDrawerContent(
+                    navHostController = navController,
+                    tabs = navigationTabs,
+                )
+            },
+        ) {
+            DestinationsNavHost(
+                startRoute = startRoute,
+                navGraph = NavGraphs.root,
+                engine = navigationEngine,
+                navController = navController,
+                manualComposableCallsBuilder = {
+                    composable(TaskListScreenDestination) {
+                        TaskListScreen(
+                            navigator = destinationsNavigator,
+                            windowSize = windowSize,
+                        )
+                    }
+                },
+            )
```

Comment:
These lines are repeated in all three nav flows, we should pull this out into its own function.